Build UI using angular-cli fails

The most relevant part of the error message is: "You are running version v6.14.4 of Node.js, which is not supported by Angular CLI v6.\nThe official Node.js version that is supported is 8.9 and greater."

Full error:

```
fatal: [cloudlaunch.sanbi.ac.za]: FAILED! => {"changed": true, "cmd": "npm run buildprod", "delta": "0:00:01.361430", "end": "2018-10-19 22:25:57.662513", "msg": "non-zero return code", "rc": 1, "start": "2018-10-19 22:25:56.301083", "stderr": "You are running version v6.14.4 of Node.js, which is not supported by Angular CLI v6.\nThe official Node.js version that is supported is 8.9 and greater.\n\nPlease visit https://nodejs.org/en/ to find instructions on how to update Node.js.\n\nnpm ERR! Linux 4.4.0-119-generic\nnpm ERR! argv \"/usr/bin/node\" \"/usr/bin/npm\" \"run\" \"buildprod\"\nnpm ERR! node v6.14.4\nnpm ERR! npm  v3.10.10\nnpm ERR! code ELIFECYCLE\nnpm ERR! cloudlaunch-ui@0.0.0 buildprod: `ng build --prod --aot=false --build-optimizer=false`\nnpm ERR! Exit status 3\nnpm ERR! \nnpm ERR! Failed at the cloudlaunch-ui@0.0.0 buildprod script 'ng build --prod --aot=false --build-optimizer=false'.\nnpm ERR! Make sure you have the latest version of node.js and npm installed.\nnpm ERR! If you do, this is most likely a problem with the cloudlaunch-ui package,\nnpm ERR! not with npm itself.\nnpm ERR! Tell the author that this fails on your system:\nnpm ERR!     ng build --prod --aot=false --build-optimizer=false\nnpm ERR! You can get information on how to open an issue for this project with:\nnpm ERR!     npm bugs cloudlaunch-ui\nnpm ERR! Or if that isn't available, you can get their info via:\nnpm ERR!     npm owner ls cloudlaunch-ui\nnpm ERR! There is likely additional logging output above.\n\nnpm ERR! Please include the following file with any support request:\nnpm ERR!     /opt/cloudlaunch-ui/npm-debug.log", "stderr_lines": ["You are running version v6.14.4 of Node.js, which is not supported by Angular CLI v6.", "The official Node.js version that is supported is 8.9 and greater.", "", "Please visit https://nodejs.org/en/ to find instructions on how to update Node.js.", "", "npm ERR! Linux 4.4.0-119-generic", "npm ERR! argv \"/usr/bin/node\" \"/usr/bin/npm\" \"run\" \"buildprod\"", "npm ERR! node v6.14.4", "npm ERR! npm  v3.10.10", "npm ERR! code ELIFECYCLE", "npm ERR! cloudlaunch-ui@0.0.0 buildprod: `ng build --prod --aot=false --build-optimizer=false`", "npm ERR! Exit status 3", "npm ERR! ", "npm ERR! Failed at the cloudlaunch-ui@0.0.0 buildprod script 'ng build --prod --aot=false --build-optimizer=false'.", "npm ERR! Make sure you have the latest version of node.js and npm installed.", "npm ERR! If you do, this is most likely a problem with the cloudlaunch-ui package,", "npm ERR! not with npm itself.", "npm ERR! Tell the author that this fails on your system:", "npm ERR!     ng build --prod --aot=false --build-optimizer=false", "npm ERR! You can get information on how to open an issue for this project with:", "npm ERR!     npm bugs cloudlaunch-ui", "npm ERR! Or if that isn't available, you can get their info via:", "npm ERR!     npm owner ls cloudlaunch-ui", "npm ERR! There is likely additional logging output above.", "", "npm ERR! Please include the following file with any support request:", "npm ERR!     /opt/cloudlaunch-ui/npm-debug.log"], "stdout": "\n> cloudlaunch-ui@0.0.0 buildprod /opt/cloudlaunch-ui\n> ng build --prod --aot=false --build-optimizer=false", "stdout_lines": ["", "> cloudlaunch-ui@0.0.0 buildprod /opt/cloudlaunch-ui", "> ng build --prod --aot=false --build-optimizer=false"]}
```

This is in `roles/cloudlaunch/tasks/setup_webui.yml`.

Thanks @nuwang - but now I get the error: "BuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):\nError: Cannot find module 'node-sass'\n"

I'll try and debug that later.

Complete error:

```
TASK [cloudlaunch : Build UI using angular-cli] ********************************************************************************************************************************************************************************************************************************
fatal: [cloudlaunch.sanbi.ac.za]: FAILED! => {"changed": true, "cmd": "npm run buildprod", "delta": "0:01:27.996868", "end": "2018-10-22 14:12:58.584346", "msg": "non-zero return code", "rc": 1, "start": "2018-10-22 14:11:30.587478", "stderr": "\nERROR in ./src/assets/scss/angular2-material-custom-theme.scss\nModule build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):\nModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):\nError: Cannot find module 'node-sass'\n    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)\n    at Function.Module._load (internal/modules/cjs/loader.js:506:25)\n    at Module.require (internal/modules/cjs/loader.js:636:17)\n    at require (internal/modules/cjs/helpers.js:20:18)\n    at Object.sassLoader (/opt/cloudlaunch-ui/node_modules/sass-loader/lib/loader.js:46:72)\n    at runLoaders (/opt/cloudlaunch-ui/node_modules/webpack/lib/NormalModule.js:286:20)\n    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:364:11\n    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:230:18\n    at runSyncOrAsync (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:143:3)\n    at iterateNormalLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:229:2)\n    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:202:4\n    at VirtualFileSystemDecorator.readFile (/opt/cloudlaunch-ui/node_modules/@ngtools/webpack/src/virtual_file_system_decorator.js:47:13)\n    at processResource (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:199:11)\n    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:158:10)\n    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)\n    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:173:18\n    at loadLoader (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/loadLoader.js:36:3)\n    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:169:2)\n    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)\n    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:173:18\n    at loadLoader (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/loadLoader.js:36:3)\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! cloudlaunch-ui@0.0.0 buildprod: `ng build --prod --aot=false --build-optimizer=false`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the cloudlaunch-ui@0.0.0 buildprod script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /root/.npm/_logs/2018-10-22T14_12_58_534Z-debug.log", "stderr_lines": ["", "ERROR in ./src/assets/scss/angular2-material-custom-theme.scss", "Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):", "ModuleBuildError: Module build failed (from ./node_modules/sass-loader/lib/loader.js):", "Error: Cannot find module 'node-sass'", "    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)", "    at Function.Module._load (internal/modules/cjs/loader.js:506:25)", "    at Module.require (internal/modules/cjs/loader.js:636:17)", "    at require (internal/modules/cjs/helpers.js:20:18)", "    at Object.sassLoader (/opt/cloudlaunch-ui/node_modules/sass-loader/lib/loader.js:46:72)", "    at runLoaders (/opt/cloudlaunch-ui/node_modules/webpack/lib/NormalModule.js:286:20)", "    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:364:11", "    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:230:18", "    at runSyncOrAsync (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:143:3)", "    at iterateNormalLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:229:2)", "    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:202:4", "    at VirtualFileSystemDecorator.readFile (/opt/cloudlaunch-ui/node_modules/@ngtools/webpack/src/virtual_file_system_decorator.js:47:13)", "    at processResource (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:199:11)", "    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:158:10)", "    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)", "    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:173:18", "    at loadLoader (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/loadLoader.js:36:3)", "    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:169:2)", "    at iteratePitchingLoaders (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:165:10)", "    at /opt/cloudlaunch-ui/node_modules/loader-runner/lib/LoaderRunner.js:173:18", "    at loadLoader (/opt/cloudlaunch-ui/node_modules/loader-runner/lib/loadLoader.js:36:3)", "npm ERR! code ELIFECYCLE", "npm ERR! errno 1", "npm ERR! cloudlaunch-ui@0.0.0 buildprod: `ng build --prod --aot=false --build-optimizer=false`", "npm ERR! Exit status 1", "npm ERR! ", "npm ERR! Failed at the cloudlaunch-ui@0.0.0 buildprod script.", "npm ERR! This is probably not a problem with npm. There is likely additional logging output above.", "", "npm ERR! A complete log of this run can be found in:", "npm ERR!     /root/.npm/_logs/2018-10-22T14_12_58_534Z-debug.log"], "stdout": "\n> cloudlaunch-ui@0.0.0 buildprod /opt/cloudlaunch-ui\n> ng build --prod --aot=false --build-optimizer=false\n\n\nDate: 2018-10-22T14:12:58.396Z\nHash: 987471d6153f24a5e218\nTime: 83599ms\nchunk {scripts} scripts.246ce29d36f8e8f42ed1.js (scripts) 82 kB  [rendered]\nchunk {0} runtime.5771c6cbeea81a4c665b.js (runtime) 2.31 kB [entry] [rendered]\nchunk {1} 1.012832af50e5fd1fe94e.js () 116 kB  [rendered]\nchunk {2} main.8da1f0ca97bee2616f53.js (main) 1.74 MB [initial] [rendered]\nchunk {3} polyfills.c77ef8630fa31234f495.js (polyfills) 59.7 kB [initial] [rendered]\nchunk {4} styles.38795fe50c7e84fe79c2.css (styles) 3.05 kB [initial] [rendered]\nchunk {5} 5.e0d15bf9b83abc66563c.js () 90.7 kB  [rendered]\nchunk {6} 6.ef6a158807e750c160ee.js () 63 kB  [rendered]\nchunk {7} 7.2829eb48ac81360d1994.js () 3.78 kB  [rendered]\n\nWARNING in ./src/app/catalog/components/appliance-detail-control/app-placeholder.component.ts 80:8-59\nSystem.import() is deprecated and will be removed soon. Use import() instead.\nFor more info visit https://webpack.js.org/guides/code-splitting/", "stdout_lines": ["", "> cloudlaunch-ui@0.0.0 buildprod /opt/cloudlaunch-ui", "> ng build --prod --aot=false --build-optimizer=false", "", "", "Date: 2018-10-22T14:12:58.396Z", "Hash: 987471d6153f24a5e218", "Time: 83599ms", "chunk {scripts} scripts.246ce29d36f8e8f42ed1.js (scripts) 82 kB  [rendered]", "chunk {0} runtime.5771c6cbeea81a4c665b.js (runtime) 2.31 kB [entry] [rendered]", "chunk {1} 1.012832af50e5fd1fe94e.js () 116 kB  [rendered]", "chunk {2} main.8da1f0ca97bee2616f53.js (main) 1.74 MB [initial] [rendered]", "chunk {3} polyfills.c77ef8630fa31234f495.js (polyfills) 59.7 kB [initial] [rendered]", "chunk {4} styles.38795fe50c7e84fe79c2.css (styles) 3.05 kB [initial] [rendered]", "chunk {5} 5.e0d15bf9b83abc66563c.js () 90.7 kB  [rendered]", "chunk {6} 6.ef6a158807e750c160ee.js () 63 kB  [rendered]", "chunk {7} 7.2829eb48ac81360d1994.js () 3.78 kB  [rendered]", "", "WARNING in ./src/app/catalog/components/appliance-detail-control/app-placeholder.component.ts 80:8-59", "System.import() is deprecated and will be removed soon. Use import() instead.", "For more info visit https://webpack.js.org/guides/code-splitting/"]}
```

--- a/README.md
+++ b/README.md
@@ -27,8 +27,14 @@ content:
 
 ```
 cloudlaunch_admin_password: "CHANGEMEONINSTALL"
+cloudlaunch_admin_email: "CHANGETOADMINEMAIL"
 dbpassword: "CHANGEMEONINSTALL"
 server_name: beta.launch.usegalaxy.org
+skey: "CHANGEONINSTALL"
+fernet_keys: "CHANGEONINSTALL"
+# need to specify destination for certificate and key if not using letencrypt
+ssl_cert_path: /etc/nginx/ssl/cl.crt"
+ssl_key_path: /etc/nginx/ssl/cl.key"
 # These files need to be placed into roles/cloudlaunch/files/secret and vaulted
 ssl_cert_files:
   - beta.launch.usegalaxy.org_certchain.pem

--- a/roles/cloudlaunch/tasks/main.yml
+++ b/roles/cloudlaunch/tasks/main.yml
@@ -8,6 +8,7 @@
   tags: broker
 
 - include: setup_certbot.yml
+  when: ssl_cert_path is not defined
   tags: certbot
 
 - include: setup_webapp.yml

--- a/roles/cloudlaunch/tasks/setup_webapp.yml
+++ b/roles/cloudlaunch/tasks/setup_webapp.yml
@@ -35,6 +35,7 @@
     state: latest
     virtualenv_python: python3.5
 
+
 - name: Install server requirements
   pip:
     requirements: "{{ server_install_dir }}/requirements.txt"
@@ -42,6 +43,25 @@
     chdir: "{{ server_install_dir }}"
     state: latest
     virtualenv_python: python3.5
+  tags:
+    - install
+    - deploy
+
+# the psycopg2 module is needed by ansible postgres modules
+- name: Install psycopg2 (system)
+  apt:
+    name: python-psycopg2
+    state: latest
+  when: ansible_python['version']['major'] == 2
+  tags:
+    - install
+    - deploy
+
+- name: Install psycopg2 (system)
+  apt:
+    name: python3-psycopg2
+    state: latest
+  when: ansible_python['version']['major'] == 3
   tags:
     - install
     - deploy
@@ -77,6 +97,7 @@
     path: "{{ server_install_dir }}/django-cloudlaunch/cloudlaunchserver/settings.py"
     regexp: "'dsn':"
     line: "    'dsn': '{{ sentry_dsn }}'"
+  when: sentry_dsn is defined
   tags:
     - install
     - deploy
@@ -130,14 +151,14 @@
 - name: Set server secret key
   replace:
     path: "{{ server_install_dir }}/django-cloudlaunch/cloudlaunchserver/settings.py"
-    regexp: "(CHANGEthisONinstall|{{ old_skey }})"
+    regexp: "(CHANGEthisONinstall|{{ old_skey | default('NOTHINGHERE') }})"
     replace: "{{ skey }}"
 
 - name: Set fernet keys
   lineinfile:
     path: "{{ server_install_dir }}/django-cloudlaunch/cloudlaunchserver/settings.py"
     regexp: "^FERNET_KEYS="
-    line: "FERNET_KEYS={{ fernet_keys }}"
+    line: "FERNET_KEYS='{{ fernet_keys }}'"
 
 - name: Change ownership of cloudlaunch server to web user
   file:

